test/lsp-io-tests.el

```diff
@@ -58,5 +58,18 @@
 		(should (equal (lsp--parser-read p "\"somedata\":2}")
 									 '("{\"somedata\":2}")))))
 
+(ert-deftest lsp--parser-read--multiple-chunks-ignored ()
```

Comment:
Thanks for adding this test! Would you mind adding more tests, e.g. with the `Content-Length` header itself split in multiple chunks, or the `\r\n` in two chunks?